viewing_party/party.py

```diff
@@ -1,23 +1,257 @@
+    
+    result_list = []
+
+    for movie_title in friends_unique_movies_list:
+        for movie in friends_movies:
+            if movie["title"] == movie_title:
+                result_list.append(movie) 
+                break
+
+    return result_list
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+    friends_list = user_data["friends"]
+    user_watched = user_data["watched"]
+    user_watched_titles = []
+
+    # Iterate through user's watched dicts and get the titles user has watched
+    for i in range(len(user_watched)):
+        user_watched_titles.append(user_watched[i]["title"])
+
+    movie_recs = []
+    for dict in friends_list:
+        for value in dict.values():
+            for movie in value:
+                friends_title = movie["title"]
+
+                for user_dict in user_watched:
+                    if friends_title not in user_watched_titles and movie["host"] in user_data["subscriptions"] and movie not in movie_recs:
+                        movie_recs.append(movie)
+    
+    return movie_recs
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    # New dict will hold the most watched genre
+    user_watched_list = user_data["watched"]
+    most_watched_genre = {}
+    for movie in user_watched_list:
+        value_genre = movie["genre"]
+        if value_genre in most_watched_genre:
+            most_watched_genre[value_genre] += 1
+        else:
+            most_watched_genre[value_genre] = 1
+
+    # Max_watched_genre hold the value of the most watched genre by user
+    max_watched_genre = ""
+    max_watched = 0
+    for genre, num in most_watched_genre.items():
+        if num > max_watched:
+            max_watched = num
+            max_watched_genre = genre
+    
+    # Create a dict that'll hold the titles of user's watched movies
+    user_watched_movies = {}
+    for movie in user_watched_list:
+        key = movie["title"]
+        user_watched_movies[key] = 1
+
+    # List of recommended movies by most watched genre and user hasn't watched it.
+    rec_movies_by_gender = {}
+    friends_list = user_data["friends"]
+    for dict in friends_list:
+        for value in dict.values():
+            for movie in value:
+                watched_by_friend = movie["title"]
+                genre_movie_friend = movie["genre"]
+                # La agrego si yo no la he visto y si es del genero que mas veo
+                if watched_by_friend in rec_movies_by_gender:
+                    continue
+                if watched_by_friend not in user_watched_movies and genre_movie_friend == max_watched_genre:
+                    rec_movies_by_gender[watched_by_friend] = movie
+
+    return list(rec_movies_by_gender.values())

```

Comment:
How would this function change if we used `get_most_watched_genre` as a helper function?